api/agent/drivers/docker/cookie.go

```diff
@@ -346,14 +361,22 @@ func (c *cookie) PullImage(ctx context.Context) error {
 		return models.NewAPIError(code, fmt.Errorf("Failed to pull image '%s': %s", c.task.Image(), msg))
 	}
 
+	img, err := c.inspectImage(ctx, c.task.Image())
+
 	if c.drv.imgCache != nil && c.cachedImg == nil {
-		img, err := c.drv.docker.InspectImage(ctx, c.task.Image())
-		// Log and ignore these errors. Hard failure will be caught in CreateContainer()
 		if err == docker.ErrNoSuchImage {
-			log.WithError(err).Errorf("InspectImage image likely removed due to disk pressure %s", c.task.Image())
-		} else if err != nil {
-			log.WithError(err).Errorf("Failed to inspect image %s", c.task.Image())
-		} else {
+			return models.ErrCallTimeoutServerBusy
```

Comment:
this makes err no such image a call timeout now? this error seems misleading/incorrect in this case, previously it would result in a 503 internal server error (which is true, because it means we pulled an image and then it was gone - that's on us)

Reply:
well, says 'server busy' - i guess this is saying that we're low on resources, which is the case we use for ram and cpu being at capacity, which makes sense even if slightly nuanced... think you can ignore me and my latent understanding